document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -131,11 +139,16 @@ private BasicDBObject prepareUpsert(Key key, Document document) throws JsonProce
     JsonNode sanitizedJsonNode = recursiveClone(jsonNode, this::encodeKey);
     BasicDBObject setObject = BasicDBObject.parse(MAPPER.writeValueAsString(sanitizedJsonNode));
     setObject.put(ID_KEY, key.toString());
+    setObject.put(LAST_UPDATED_TIME, System.currentTimeMillis());
```

Comment:
Please call System.currentTimeMillis once and use it both the places. Based on the current code, there is a possibility that lastUpdatedTime and createTime for newly created documents will mismatch.

Also, if we plan to use the lastUpdatedTime, do we still need the older lastUpdateTime field?

Reply:
we will need to keep the older _lastUpdateTime for now because copying the time to lastUpdatedTime from _lastUpdateTime is a manual process. We need to give time to allow for this migration before deleting the older field.